### PR TITLE
Refactor autologging utilities for future modularity / extensibility

### PR DESCRIPTION
**Pull request by dbczumar**
## What changes are proposed in this pull request?

Problem: the `mlflow.utils.autologging_utils` module is huge. As part of an effort to introduce a `silent` option to `autolog()`, we need to add even more autologging utility functions. To avoid compounding the large module problem, this PR moves `mlflow/utils/autologging_utils` to `mlflow/utils/autologging/__init__`; this will allow us to add more autologging-related utilities as new module file(s) in the `mlflow/utils/autologging/`, rather than dropping them all in the same `autologging_utils` file.

## How is this patch tested?

**NOTE THAT WINDOWS & PYTHON LARGE / SMALL TESTS CURRENTLY FAIL DUE TO AN UNRELATED SQLALCHEMY ISSUE (https://github.com/mlflow/mlflow/pull/4180) THAT SHOULD BE FIXED BEFORE MERGING** 

- Existing unit tests (this is just some refactoring)

### Is this a user-facing change?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/server-infra`: MLflow server, JavaScript dev server
- [X] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [X] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/utils/autologging/__init__.py [hunk 1]
```diff
@@ -110,13 +110,10 @@ def _update_wrapper_extended(wrapper, wrapped):
     return updated_wrapper
```
Comment (dbczumar) [inline mlflow/utils/autologging/__init__.py:110]: Move `mlflow/utils/autologging_utils.py` to `mlflow/utils/autologging/__init__.py`, allowing us to add autologging utility modules as separate file(s) in the new `mlflow/utils/autologging` directory in the future.

Also address a TODO in the docstring of the `wrap_patch` function.

File: mlflow/utils/autologging/__init__.py [hunk 2]
```diff
@@ -315,7 +312,7 @@ def _check_version_in_range(ver, min_ver, max_ver):
 
 
 def _load_version_file_as_dict():
-    version_file_path = resource_filename(__name__, "../ml-package-versions.yml")
+    version_file_path = resource_filename(__name__, "../../ml-package-versions.yml")
```
Comment (dbczumar) [inline mlflow/utils/autologging/__init__.py:315]: As a result of https://github.com/mlflow/mlflow/pull/4184/files#r596149139, `ml-package-versions.yml` is now an additional directory above this module.

**Comment by apurva-koti** [comment]
Let's file a follow up to a few things out of `__init__` into (some) new file(s)! Agree the module is huge and could use a readability boost.

File: mlflow/utils/autologging/__init__.py [hunk 3]
```diff
@@ -110,13 +110,10 @@ def _update_wrapper_extended(wrapper, wrapped):
     return updated_wrapper
 
 
-def wrap_patch(destination, name, patch, settings=None):
+def _wrap_patch(destination, name, patch, settings=None):
```
Comment (mohamad-arabi) [inline mlflow/utils/autologging/__init__.py:113]: I just confirmed on master branch that wrap_patch is not called outside of autologging_utils and test_autologging_utils files

**Review by mohamad-arabi — APPROVED** [review]
LGTM! Like you said make sure sqlalchemy issue is resolved first. Need to insure a green build before a big module change:
`mlflow.utils.autologging_utils` `->` `mlflow.utils.autologging`

**Pull request merged**